src/main/java/net/earthcomputer/clientcommands/features/VillagerRngSimulator.java

```diff
@@ -0,0 +1,538 @@
+            onBadRNG("ambient");
+            justReset = true;
+        }
+
+        if (totalAmbientSounds == 0) {
+            totalAmbientSounds++;
+            firstPitch = pitch;
+            ambientSoundTime = -80;
+            if (!justReset) {
+                ClientCommandHelper.addOverlayMessage(getCrackedState().getMessage(false).withStyle(ChatFormatting.RED), 100);
+            }
+            return;
+        }
+
+        if (totalAmbientSounds == 1) {
+            totalAmbientSounds++;
+            ticksBetweenSounds = ambientSoundTime - (-80);
+            secondPitch = pitch;
+            ambientSoundTime = -80;
+
+            if (seedsFromTwoPitches != null) {
+                int matchingSeeds = 0;
+                long matchingSeed = 0;
+                nextSeed: for (long seed : seedsFromTwoPitches) {
+                    JRand rand = JRand.ofInternalSeed(seed);
+                    rand.nextInt(100);
+                    for (int i = -80; i < ticksBetweenSounds - 80 - 1; i++) {
+                        if (rand.nextInt(1000) < i) {
+                            continue nextSeed;
+                        }
+                        rand.nextInt(100);
+                    }
+                    if (rand.nextInt(1000) >= ticksBetweenSounds - 80 - 1) {
+                        continue;
+                    }
+                    float simulatedThirdPitch = (rand.nextFloat() - rand.nextFloat()) * 0.2f + 1.0f;
+                    if (simulatedThirdPitch == pitch) {
+                        matchingSeeds++;
+                        matchingSeed = rand.getSeed();
+                    }
+                }
+                seedsFromTwoPitches = null;
+                if (matchingSeeds == 1) {
+                    random = JRand.ofInternalSeed(matchingSeed);
+                    random.nextInt(100);
+                    ClientCommandHelper.addOverlayMessage(Component.translatable("commands.cvillager.crack.success", Long.toHexString(matchingSeed)).withStyle(ChatFormatting.GREEN), 100);
+                    return;
+                }
+            }
+
+            long[] seeds = crackSeed();
+            if (seeds.length == 1) {
+                random = JRand.ofInternalSeed(seeds[0]);
+                random.nextInt(100);
+                ClientCommandHelper.addOverlayMessage(Component.translatable("commands.cvillager.crack.success", Long.toHexString(seeds[0])).withStyle(ChatFormatting.GREEN), 100);
+            } else {
+                totalAmbientSounds = 1;
+                firstPitch = pitch;
+                secondPitch = Float.NaN;
+                seedsFromTwoPitches = seeds.length > 0 ? seeds : null;
+                ambientSoundTime = -80;
+                ClientCommandHelper.addOverlayMessage(Component.translatable("commands.cvillager.crack.failed", seeds.length).withStyle(ChatFormatting.RED), 100);
+            }
+        }
+    }
+
+    public void onBadRNG(@Translatable(prefix = "villagerManip.reset.") String reason) {
```

Comment:
Maybe use an `enum` as opposed to string literals as raw strings are more error-prone.

Reply:
this is how it works on player rng gen, if I were to change it, shouldn't I change it there too?

Reply:
Sure yeah.

Reply:
ok, i'll need permission from @Earthcomputer to do that tho for sure

Reply:
I think strings are fine here because mcdev helps with the completion and will give warnings if you type a translation which doesn't exist

Reply:
Alright. Note that two still slipped through though, and I also noticed there's one in `PlayerRandCracker` (`resetCracker("enchanted")`). Refactoring (renaming) doesn't seem to work well.

Reply:
That's a bug in mcdev that needs to be fixed then